src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs

```diff
@@ -552,6 +552,30 @@ internal DocumentState UpdateTree(SyntaxNode newRoot, PreservationMode mode)
                 textSource: text,
                 LoadTextOptions,
                 treeSource: new ConstantValueSource<TreeAndVersion>(treeAndVersion));
+
+            // use static method so we don't capture references to this
+            static (ITextAndVersionSource, TreeAndVersion) CreateTreeWithLazyText(
```

Comment:
this method became a nested helper.  which really screwed up github diff.  the above method ends up looking like a move.  it is otherwise untouched by me.